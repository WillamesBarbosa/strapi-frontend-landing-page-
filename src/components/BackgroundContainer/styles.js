import styled, { css } from 'styled-components';

const handleChangeBackgroundColor = (theme) => css`
  background-color: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, backgroundColor }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${backgroundColor && handleChangeBackgroundColor(theme)}
    min-height: 100vh;
    padding: 5rem;
  `}
`;
