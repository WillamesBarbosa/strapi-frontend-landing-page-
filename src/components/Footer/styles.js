import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.size.small};
    }
    text-align: center;
  `}
`;
