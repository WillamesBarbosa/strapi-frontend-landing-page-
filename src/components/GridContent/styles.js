import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.font.spacings.xbig} 0;
  `}
`;
