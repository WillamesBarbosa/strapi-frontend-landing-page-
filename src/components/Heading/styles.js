import styled, { css } from 'styled-components';

const titleSize = {
  xxsmall: (theme) =>
    css`
      font-size: ${theme.font.size.xxsmall};
    `,
  xsmall: (theme) =>
    css`
      font-size: ${theme.font.size.xsmall};
    `,
  small: (theme) =>
    css`
      font-size: ${theme.font.size.small};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.size.medium};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.size.big};
    `,
  xbig: (theme) =>
    css`
      font-size: ${theme.font.size.xbig};
    `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme) => css`
  @media ${theme.mediaQuerie.mediumQuerie} {
    font-size: ${theme.font.size.medium};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
    ${mediaFont(theme)}
  `}
`;
