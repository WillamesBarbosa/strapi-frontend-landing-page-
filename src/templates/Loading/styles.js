import styled, { css, keyframes } from 'styled-components';
const isRotating = () => keyframes`
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  `;

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: ${theme.colors.primaryColor};
  `}
`;

export const loadingBar = styled.div`
  ${({ theme }) => css`
    position: absolute;
    width: 20rem;
    height: 20rem;
    border: 1rem solid white;
    border-top: 1rem solid ${theme.colors.secondaryColor};
    border-radius: 50%;
    animation: ${isRotating} 0.8s infinite linear;
  `}
`;
