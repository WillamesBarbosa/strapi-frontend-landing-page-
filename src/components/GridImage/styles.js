import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    margin: auto;
    width: 85%;
    ${TextComponent} {
      margin-bottom: ${theme.font.spacings.xbig};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.big};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;

    &:hover {
      transform: scale(1.2);
    }
  `}
`;
