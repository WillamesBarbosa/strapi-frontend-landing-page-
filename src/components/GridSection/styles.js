import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.font.spacings.xbig};
      margin-top: ${theme.font.spacings.xbig};
      font-size: ${theme.font.spacings.small};

      & > ${HeadingContainer} {
        font-size: ${theme.font.size.small};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.big};
    overflow: hidden;
    margin: auto;
    width: 80%;
    @media ${theme.mediaQuerie.mediumQuerie} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      font-size: ${theme.font.size.medium};
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
