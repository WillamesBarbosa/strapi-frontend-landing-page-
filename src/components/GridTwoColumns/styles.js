import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.font.spacings.big};
    padding: ${theme.font.spacings.big};
    width: 85%;
    margin: auto;

    @media ${theme.mediaQuerie.mediumQuerie} {
      grid-template-columns: 1fr;
    }

    & ${Title} {
      margin-bottom: ${theme.font.spacings.big};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.mediaQuerie.mediumQuerie} {
      margin-bottom: ${theme.font.spacings.big};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
