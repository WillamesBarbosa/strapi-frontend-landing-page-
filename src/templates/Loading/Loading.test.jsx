import { renderTheme } from '../../styles/render-theme';
import { Loading } from '.';

describe('<Loading />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Loading />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
