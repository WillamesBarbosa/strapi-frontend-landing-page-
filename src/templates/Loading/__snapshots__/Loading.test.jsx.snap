// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0A1128;
}

.c1 {
  position: absolute;
  width: 20rem;
  height: 20rem;
  border: 1rem solid white;
  border-top: 1rem solid #dc143c;
  border-radius: 50%;
  -webkit-animation: hEJTsP 0.8s infinite linear;
  animation: hEJTsP 0.8s infinite linear;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;
